go/pkg/client/tree_test.go

```diff
@@ -490,6 +505,92 @@ func TestComputeMerkleTree(t *testing.T) {
 				TotalInputBytes:  2*fooDg.Size + fooDirDg.Size,
 			},
 		},
+		{
+			desc: "File relative symlink (preserved)",
+			input: []*inputPath{
+				{path: "fooDir/foo", fileContents: fooBlob, isExecutable: true},
+				{path: "fooSym", isSymlink: true, symlinkTarget: "fooDir/foo"},
+			},
+			spec: &command.InputSpec{
+				// The symlink target will be traversed recursively.
+				Inputs: []string{"fooSym"},
+			},
+			rootDir: &repb.Directory{
+				Directories: []*repb.DirectoryNode{{Name: "fooDir", Digest: fooDirDgPb}},
+				Symlinks:    []*repb.SymlinkNode{{Name: "fooSym", Target: "fooDir/foo"}},
+			},
+			additionalBlobs: [][]byte{fooBlob, fooDirBlob},
+			wantCacheCalls: map[string]int{
+				"fooDir/foo": 1,
+				"fooSym":     1,
+			},
+			wantStats: &client.TreeStats{
+				InputDirectories: 2,
+				InputFiles:       1,
+				InputSymlinks:    1,
+				TotalInputBytes:  fooDg.Size + fooDirDg.Size,
```

Comment:
This seems low... maybe not counting the symlink nodes? Ditto below.

Reply:
Hmm, you mean the `TotalInputBytes` should count the bytes for symlinks? I didn't do that because 1) the symlink node itself isn't associated with a `Digest`, and 2) will counting in both the symlink and the target result in double counting?